packages/component-library/src/components/overlay/mt-modal/sub-components/mt-modal-root.vue

```diff
@@ -0,0 +1,63 @@
+<template>
```

Comment:
Just for interest: what is the benefit of having separate components for root, trigger, inset, etc. instead of one modal component with `isOpen` property?

Reply:
The idea was to provide simpler developer experience. You still are able to use an isOpen prop in cases where you need that. For example when the modal opens instantly when the page loads or when you want to programatically show the modal let's say you want to show it 30 minutes after the user has been inactive. But for the 99% of cases a modal gets opened when the user clicks on a button that looks like a link or a normal button and the user and a modal will almost always have an cancel and a confirm button. Those are just tiny abstractions that won't add up to much bundle size. If someones does not prefer to use them they can make use of the `isOpen` prop and the other components will be tree-shaked. 

An example: 

```vue
<mt-modal-root>
  <mt-modal>
    <template #default>
      <p>My content</p>
    </template>
    
    <template #footer>
       <mt-modal-close :as="MtButton" variant="secondary">Close</mt-modal-close>
    
       <mt-modal-action 
         :as="MtButton"
         variant="primary"
         @click="async (close) => {
           await makeRequest();
         
           close();
         }"
         >Confirm</mt-modal-close>
    </template>
  <mt-modal>
</mt-modal-root>
```

There's no need to define an isOpen property, if not wanted.

Opposed to the work a developer has to perform in platform right now:

```vue
<template>
  <sw-modal v-if="isOpen">
    <template #default>
      <p>My content</p>
    </template>

    <template #footer>
      <sw-button>Close</sw-button>

      <sw-button
        @click="await () => {
          async makeRequest();
          isOpen = false;
         }"
      >
        Confirm
      </sw-button>
    </template>
  </sw-modal>
</template>

<script lang="ts">
export default {
  data() {
    return {
      isOpen: false,
    };
  },
};
</script>
```

I know the difference is not much, but I still believe that doing many little improvements in developer experience add up quickly.